al_log_filter.js

```diff
@@ -38,10 +40,24 @@ var initJsonFilter = function(filter) {
  *  @NOTE: If json filter  is bad then 'messages' is returned unfiltered.
  */
 
-var filterJson = function(messages, filter) {
+var filterJson = function (messages, filter) {
     const filterJ = initJsonFilter(filter);
     if (filterJ) {
-        return where(messages, filterJ);
+        let result = [];
+        let clonedMsgs = lodashcloneDeep(messages);
+        if (Array.isArray(filterJ)) {
+            filterJ.forEach(function (e) {
+                lodashRemove(clonedMsgs, function (t) {
+                    if (lodashFilter([t], e).length) {
+                        result.push(t);
+                        return true;
+                    }
+                });
+            });
+        } else {
+            result = result.concat(lodashFilter(clonedMsgs, filterJ));
```

Comment:
why do we need `concat()` here?

Reply:
lodash filter returns array so we need to contact it with our defined result array.

Reply:
lodash.filter returns array so we need to concat it with our defined result array.